### PR TITLE
Fix issues with unitialised variables

### PR DESCRIPTION
**Pull request by russfiedler**
Addresses issues here. https://github.com/mom-ocean/MOM5/issues/349

### CODE REVIEW

**Comment by aekiss** [comment]
@russfiedler this looks good to me - should I merge?

**Comment by aidanheerdegen** [comment]
The `fms` stuff should be fixed in the FMS repo

**Comment by aidanheerdegen** [comment]
Specifically the changes under `shared`. Sorry I don't make the rules .

File: src/shared/exchange/xgrid.F90 [hunk 1]
```diff
@@ -477,9 +477,15 @@ module xgrid_mod
  integer :: id_get_2_from_xgrid = 0
  integer :: id_put_2_to_xgrid = 0
  integer :: id_setup_xmap = 0
- integer :: id_load_xgrid1, id_load_xgrid2, id_load_xgrid3
- integer :: id_load_xgrid4, id_load_xgrid5
- integer :: id_load_xgrid, id_set_comm, id_regen, id_conservation_check
+ integer :: id_load_xgrid = 0
```
Comment (aidanheerdegen) [inline src/shared/exchange/xgrid.F90:480]: Another option for these is to only call the timer functions if `xgrid_clocks_on` is set, e.g
```fortran
if(xgrid_clocks_on) call mpp_clock_begin(id_load_xgrid1)
```

Comment (aidanheerdegen) [reply src/shared/exchange/xgrid.F90:480]: I see `app_clock_begin` returns if the `id == 0`, so yeah, same same.

**Comment by aidanheerdegen** [comment]
@russfiedler can you rebase this on `master` as that fixes the FMS stuff. If not I can take a crack at it.

**Comment by aidanheerdegen** [comment]
Now I'm confused, as I did fix that. Sorry for the spam, I'll take another look and report back.

**Comment by aidanheerdegen** [comment]
Ok, rebased on top of the FMS update which actually did fix the clock id variable initialisation and pushed to the PR branch. Hope that is ok @russfiedler.

**Comment by aidanheerdegen** [comment]
Okey-doke, this PR no longer touches FMS, so should be fine to merge.

**Pull request merged**